pubsub/apiserver/messages.go

```diff
@@ -33,3 +35,34 @@ type Details struct {
 	Servers   map[string]APIServer `yaml:"servers"`
 	LocalOnly bool                 `yaml:"local-only"`
 }
+
+// ConnectTopic is the topic name for the published message
+// whenever an agent conntects to the API server.
+const ConnectTopic = "apiserver.agent-connect"
+
+// DisconnectTopic is the topic name for the published message
+// whenever an agent disconntects to the API server.
+const DisconnectTopic = "apiserver.agent-disconnect"
+
+// APIConnection holds all the salient pieces of information that are
+// available when an agent connects to the API server.
+type APIConnection struct {
+	AgentTag        string `yaml:"agent-tag"`
+	ControllerAgent bool   `yaml:"controller-agent,omitempty"`
+	ModelUUID       string `yaml:"model-uuid"`
+	ConnectionID    uint64 `yaml:"connection-id"`
+	Origin          string `yaml:"origin"`
+	UserData        string `yaml:"user-data,omitempty"`
+}
+
+const PresenceRequestTopic = "presense.request"
```

Comment:
presence.request (and ditto below)

Reply:
Ugh, now I've looked at it too much and presense looks right.

Reply:
I'll remove these from this branch. They are only needed for the presence-worker branch.